### PR TITLE
Feature: loading screen when auth

### PR DESCRIPTION
**Pull request by kakoc**
Added LoadingScreen component and WithLoader HOC for better separation of responsibility. 
Also integrated loader into process of retrieving session data. If you will say that it is a good idea i can improve other parts.

P.S. i am learning TS and so there are some places where i used <any> and other bad practices because now there is not enough knowledge. I will improve these parts in the feature.

### CODE REVIEW

**Comment by dzmitry-varabei** [comment]
Thank you for this PR! 👍 
Could you please replace animated RS logo with something like this - https://codepen.io/matt-tonks/pen/YjvPay

**Comment by kakoc** [comment]
loading-demo

Something like that. I think that this is combined with the current design of the app.

**Comment by dzmitry-varabei** [comment]
The new animation is great! :)

File: src/app/components/HOCS/WithLoader.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import LoadingScreen from '../LoadingScreen';
+
+const WithLoader = (WrappedComponent: React.ComponentType) => (props: any) => (
+    <React.Fragment>
+        <LoadingScreen show={props.isLoading} />
+        <WrappedComponent {...props} />
+    </React.Fragment>
+);
+
+export default WithLoader;
```
Comment (R1ZZU) [inline src/app/components/HOCS/WithLoader.tsx]: I'd rather move `HOCS` folder to `src/app/`, but that's arguable

File: src/app/core/reducers/user.ts [hunk 2]
```diff
@@ -34,16 +36,51 @@ export function userReducer(state = initialState, action: Action<any>): UserStat
...
+            };
+        }
+        case USER.FEED_FETCH_FAIL: {
+            return {
+                ...state,
+                isLoading: false,
+            };
+        }
+
+        case USER.FETCH_SESSION: {
```
Comment (R1ZZU) [inline src/app/core/reducers/user.ts]: actions could be grouped:

```javascript
...
case USER.PARTICIPATIONS_FETCH_FAIL:
case USER.FEED_FETCH:
case USER.FEED_FETCH_FAIL:
case USER.FETCH_SESSION_FAIL:
// etc
    return {
        ...state,
        isLoading: false,
    };
```

**Pull request merged**